### PR TITLE
118 Time Series Momentum Effect

### PR DESCRIPTION
**Pull request by BlueButterBall**
New write-up for Time Series Momentum Effect (Quantpedia 118: https://quantpedia.com/Screener/Details/118 )

### CODE REVIEW

File: 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    hist = self.History([symbol], 300, Resolution.Daily).loc[symbol]
+    for i in hist.itertuples():
+        self.roc[symbol].Update(i.Index, i.value)
+</pre>
+</div>
+<p>
+  In <code>OnData(self, data)</code>, indicators for all futures contracts are updated every day with the settlement price.
+</p>
+<p>
+  We choose instruments with positive 12-month momentum to long, and instrucments with negative 12-month momentum to short.
```
Comment (simonsonjack) [inline 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html]: "We take a long position in instruments with positive 12-month momentum and short positions with negative momentum."

File: 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+</p>
+<div class="section-example-container">
+    <pre class="python">
+        self.long = [symbol for symbol in self.roc if self.roc[symbol].Current.Value > 0]
+        self.short =  [symbol for symbol in self.roc if self.roc[symbol].Current.Value < 0]
+    </pre>
+</div>
+
+<p>
+    The weights are inversely proportional to the contracts' <a href="https://www.investopedia.com/terms/v/volatility.asp">volatility</a>. As we stated in the introduction, the volatility can be estimated by GARCH model or histrocial volatility. Here, for the sake of simplity, we use historical volatility here. The <code> volatility </code> here is a dictionary containging the reciprocal 252-day historical volatility.
```
Comment (simonsonjack) [inline 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html]: "As we stated in the introduction, a security's volatility can be estimated using a GARCH model or historical volatility. For the sake of simplicity, we use historical volatility. The <code> volatility </code> here is a dictionary containing the reciprocal 252-day historical volatility."

File: 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    <pre class="python">
+        hist = self.History(self.Symbol(symbol), 252, Resolution.Daily).loc[symbol]['value']
+        log_return = np.log(hist/hist.shift()).dropna().tolist()
+        volatility[symbol] = 1/np.std(log_return,ddof = 1)
+    </pre>
+</div>
+
+
+<p>
+    Then we determine the weights for every contract and then trade.
```
Comment (simonsonjack) [inline 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html]: "Then we determine the weights for every contract and place our orders."

File: 04 Strategy Library/118 Time Series Momentum Effect/01 Introduction.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<p>
+Time series momentum is related to, but different from the phenomenon known as “momentum” in the finance literature, which is primarily cross-sectional in nature.
+The momentum literature focuses on the relative performance of securities in the cross section, finding that securities that recently outperformed their peers over the past 3 to 12 months continue to outperform their peers on average over the next month.
```
[the remark this answers is not in the document]
Comment (BlueButterBall) [reply 04 Strategy Library/118 Time Series Momentum Effect/01 Introduction.html]: The momentum literature focuses on the relative performance of securities in the cross section, finding that securities **that recently outperformed their peers over the past 3 to 12 months** continue to outperform their peers on average over the next month. 

Here, "that recently outperformed their peers over the past 3 to 12 months" is a clause, so probably no need to insert "and" between "months" and "continue"?

[the remark this answers is not in the document]
Comment (BlueButterBall) [reply 04 Strategy Library/118 Time Series Momentum Effect/01 Introduction.html]: > outperformed their peers over the past 3 to 12 months and continue to outperform their peers
> 
> -- Insert 'and' between 'months' and 'continue'
The momentum literature focuses on the relative performance of securities in the cross section, finding that securities that recently outperformed their peers over the past 3 to 12 months continue to outperform their peers on average over the next month.

Here, "that recently outperformed their peers over the past 3 to 12 months" is a clause, so probably no need to insert "and" between "months" and "continue"?

[the remark this answers is not in the document]
Comment (simonsonjack) [reply 04 Strategy Library/118 Time Series Momentum Effect/01 Introduction.html]: Good point, I was reading it in the wrong context. I'd replace
`to outperform their peers` with
`to do so`.
The way it right now, the sentence repeats the same phrase twice and it stands out as a bit too wordy. Make it more concise.

File: 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html [hunk 5]
```diff
@@ -18,8 +17,8 @@
 </pre>
 </div>
 <p>
-  Here we use the indicator <code>RateOfChange(period)</code> to simulate the momentum return. Here the period is 12 months.
-  As we are using the custom data, the indicator initialization should use the history request to update the value manually.
+  Here we use a 12-month <code>RateOfChange(period)</code> indicator to simulate the momentum return. Here the period is 12 months.
```
Comment (simonsonjack) [inline 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html]: You can delete 
`Here the period is 12 months.`

It is redundant

File: 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html [hunk 6]
```diff
@@ -36,7 +35,7 @@
   In <code>OnData(self, data)</code>, indicators for all futures contracts are updated every day with the settlement price.
 </p>
 <p>
-  We choose instruments with positive 12-month momentum to long, and instrucments with negative 12-month momentum to short.
+  We take a long position in instruments with positive 12-month momentum and short positions with negative momentum.
```
Comment (simonsonjack) [inline 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html]: Change to:

`We take a long position in instruments with positive 12-month momentum and short positions in those with negative momentum.`

File: 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+self.period = 252
+self.roc = {}
+for symbol in self.symbols:
+  self.roc[symbol] = self.ROC(symbol, self.period)   #Initialize ROC indicator : ROC is short for RateofChange
+
+</pre>
+</div>
+
+<p>
+  We use hisotry request to obtain historical prices. Here, <code>history</code> is the daily returns, which will be used to calculate volatilities.
```
Comment (jaredbroad) [inline 04 Strategy Library/118 Time Series Momentum Effect/02 Method.html]: sp. `hisotry`

**Comment by jaredbroad** [comment]
@simonsonjack @BlueButterBall note this is missing the addition to the quantpedia index:
https://github.com/QuantConnect/Tutorials/blob/master/quantpedia.json#L49
